vminfo_parser/vminfo_parser.py

```diff
@@ -179,6 +227,42 @@ def print_disk_space_ranges(self: t.Self, range_counts: dict[tuple[int, int], in
             disk_range_str = f"{disk_range[0]}-{disk_range[1]}"
             print(f"{disk_range_str.ljust(32)} {count}")
 
+    def print_site_usage(self: t.Self, resource: str, dataFrame: pd.DataFrame) -> None:
+        """
+        Prints the site-wide usage of a specified resource, including Memory, CPU, Disk, or VM count.
+
+
+        Args:
+            resource (str): The type of resource to summarize. Options include "Memory", "CPU", "Disk", or "VM".
+            dataFrame (pd.DataFrame): A DataFrame containing the relevant data for the site, including resource usage metrics.
+
+        Returns:
+            None: This function does not return a value; it prints the usage information directly to the console.
+        """
+
+        print(f"Site Wide {resource} Usage")
+        print("-------------------")
+        if not dataFrame.empty:
+            if resource == "CPU":
+                cpu_usage = dataFrame["Site_CPU_Usage"].astype(int)
+                for index, row in dataFrame.iterrows():
+                    print(f"{row['Site Name']}\t\t{cpu_usage[index]} Cores")
+            if resource == "Memory":
+                memory_usage = dataFrame["Site_RAM_Usage"].round(0).astype(int)
+                for index, row in dataFrame.iterrows():
+                    print(f"{row['Site Name']}\t\t{memory_usage[index]} GB")
+            elif resource == "Disk":
+                disk_usage = dataFrame["Site_Disk_Usage"]
+                for index, row in dataFrame.iterrows():
+                    print(f"{row['Site Name']}\t\t{disk_usage[index]:.0f} TB")
+            elif resource == "VM":
+                vm_count = dataFrame["Site_VM_Count"]
+                for index, row in dataFrame.iterrows():
+                    print(f"{row['Site Name']}\t\t{vm_count[index]} VMs")
```

Comment:
This should probably be a match/case statement rather than sequential if/else statements.

If not,  at least like 250 needs to be replaced with `elif`.

Reply:
I changed this to a match as per your suggestion